week-3/alarmclock/alarmclock.js

```diff
@@ -1,19 +1,70 @@
-function setAlarm() {}
 
-// DO NOT EDIT BELOW HERE
+let countdown;
 
-var audio = new Audio("alarmsound.mp3");
+function setAlarm() {
+  const input = document.getElementById("alarmSet");
+  const heading = document.getElementById("timeRemaining");
+
+  let timeInSeconds = parseInt(input.value, 10);
+
+  if (isNaN(timeInSeconds) || timeInSeconds <= 0) {
+    alert("Please enter a valid positive number for the alarm time.");
+    return;
+  }
+
+  let minutes = Math.floor(timeInSeconds / 60);
+  let seconds = timeInSeconds % 60;
+
+  countdown = setInterval(() => {
+    seconds--;
```

Comment:
This line causes a bug -  the seconds go down to -1 and when you add an alarm, the time is the time you entered -1. Can you see why this happens?